webapp/graphite/finders/cache.py

```diff
@@ -16,13 +16,23 @@ def find_nodes(self, query):
         clean_patterns = query.pattern.replace('\\', '')
         has_wildcard = clean_patterns.find('{') > -1 or clean_patterns.find('[') > -1 or clean_patterns.find('*') > -1 or clean_patterns.find('?') > -1
 
-        # 1) CarbonLink has some hosts
-        # 2) has no wildcard
-        if CarbonLink.hosts and not has_wildcard:
+        # CarbonLink has some hosts
+        if CarbonLink.hosts:
             metric = clean_patterns
-            exists = CarbonLink.precheck(metric, query.startTime)
-            if exists:
-                metric_path = metric
-                # TODO: check any info we need to put into reader @here
-                reader = CarbonCacheReader(metric)
-                yield LeafNode(metric_path, reader)
+            # query pattern has no wildcard
+            if not has_wildcard:
+                exists = CarbonLink.precheck(metric, query.startTime)
+                if exists:
+                    metric_path = metric
+                    # TODO: check any info we need to put into reader @here
+                    reader = CarbonCacheReader(metric)
+                    yield LeafNode(metric_path, reader)
+            else:
+                # expand queries in CarbonLink
+                metrics = CarbonLink.expand_query(metric)
+                # check all metrics in same valid query range
+                exists = all((CarbonLink.precheck(m, query.startTime) for m in metrics))
```

Comment:
should this be all? That would mean this only works if all metrics are in the cache, correct?